### PR TITLE
Migrate JSON Path Assertion into JMeter core

### PR DESCRIPTION
**Pull request by artem-fedorov**

### CODE REVIEW

**Comment by pmouawad** [comment]
Hi @artem-fedorov ,
Thanks for the PR
Regarding build:
- You might have to merge the trunk as I made a fix in JMS few minutes ago.

Regarding PR itself:
- Package for assertions is org.apache.jmeter.assertions , so I think you should move the class to it
- I don't see JUnit tests for JSONAssertion , only for GUI, am I missing something ? If they are really missing it would be nice if you could add some possibly spock / groovy based or Junit only

Regards

**Comment by artem-fedorov** [comment]
@pmouawad 
I moved classes to `org.apache.jmeter.assertions` folder.

Regarding Tests: You are really missed this class `test/src/org/apache/jmeter/assertions/TestJSONPathAssertion.java`

**Review by ham1 — COMMENTED** [review]
Thank you for a very useful feature, I just had a few minor comments.

File: src/components/org/apache/jmeter/extractor/json/jsonpath/JSONPathAssertion.java [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+                } else {
+                    msg="Value expected to be '%s', but found '%s'";
+                }
+                throw new RuntimeException(String.format(msg, getExpectedValue(), objectToString(value)));
+            }
+        }
+    }
+
+    private boolean arrayMatched(JSONArray value) {
+        if (value.isEmpty() && getExpectedValue().equals("[]")) {
```
[the remark this answers is not in the document]
Comment (artem-fedorov) [reply src/components/org/apache/jmeter/extractor/json/jsonpath/JSONPathAssertion.java]: Tests covered a lot of users scenarios that were formed by users for years. I think that this scenario also covered in one of tests (for example in `TestJSONPathAssertion.testGetResult_list_*`)

File: test/src/org/apache/jmeter/assertions/TestJSONPathAssertion.java [hunk 2]
```diff
@@ -0,0 +1,383 @@
...
+
+    @Test
+    public void testIsJsonValidationBool() {
+        JSONPathAssertion instance = new JSONPathAssertion();
+        boolean result = instance.isJsonValidationBool();
+        assertEquals(false, result);
+    }
+
+    @Test
+    public void testGetResult_positive() {
```
[the remark this answers is not in the document]
Comment (artem-fedorov) [reply test/src/org/apache/jmeter/assertions/TestJSONPathAssertion.java]: It's also work fine in Java :smile:

**Pull request closed without merging**